### DISCUSSION TITLE
Can i use a negative glob in globalDependencies?

### DISCUSSION THREAD

**Discussion by scamden** [vercel/turborepo#7860] [Help]
### Summary

Use case:

I want everything in a given folder except the build directory. I'm trying the following but --summarize still shows the build folder files as cache inputs.

```
  "globalDependencies": [
    "workspaces/rig/**",
    "!workspaces/rig/build/**",
  ]
```

**Comment by scamden** [comment]
hello? anyone? bueller?

**Reply by scamden** [reply]
@gsoltis tagging someone i've seen be active just to get some eyes. feels like this should be a simple question

**Reply by scamden** [reply]
i'm feeling a little frustrated that there's not been even a tiny response here... maybe i'll open a github issue to bring up the urgency. 

I believe I'm properly attempting the negative glob example shown in the docs here: https://turbo.build/repo/docs/reference/globs

Maybe these don't apply to the globalDeps?

**Reply by anthonyshew** [reply]
Hey, sorry about the lack of response. Greg doesn't work on the project anymore, but your most recent comment here brought this back up to the top, so I'm seeing it now.

Looking at what you've written, I'm wondering if you need `globalDependencies` at all. Could you share your `turbo.json` and a little bit of information about the shape of your repository? For instance, am I safe to assume that `workspaces` is [included in your Workspace definition](https://turbo.build/repo/docs/crafting-your-repository/structuring-a-repository#declaring-directories-for-packages) using `workspaces/*`? Meaning `rig` is a package?

**Reply by scamden** [reply]
thanks for getting back!

workspaces are in the workspaces definition yes and workspaces/rig is in the dependencies list of the necessary packages. What we found is that turbo didn't bust the cache on changes to these deps unless we had an actual turbo task as a dependsOn that targeted upstream dependencies. But the rig is a package that has crucial configuration like tsconfig-base.json for example, so it's a dependency for essentially every turbo task.

**Reply by anthonyshew** [reply]
It sounding to me like you could leverage [the `inputs` array](https://turbo.build/repo/docs/reference/configuration#inputs) directly here, with something like:

```json
{
  "tasks": {
    "build": {
      "dependsOn": ["^build"],
      "inputs": ["$TURBO_DEFAULT$", "!.build/**"]
    }
  }
}
```

Does that sound right?
